src/components/views/rooms/MemberInfo.js

```diff
@@ -641,10 +641,19 @@ module.exports = createReactClass({
     },
 
     _calculateOpsPermissions: async function(member) {
+        let canDeactivate = false;
+        if (this.context.matrixClient) {
+            try {
+                canDeactivate = await this.context.matrixClient.isSynapseAdministrator();
```

Comment:
`isSynapseAdministrator` catches any errors internally, so maybe just do a null-check here? The defensive try/catch makes it look like there is a reasonable chance it will throw.

```
MatrixClient.prototype.isSynapseAdministrator = function() {
    return this.whoisSynapseUser(this.getUserId())
        .then(() => true)
        .catch(() => false);
};
```